// npm-audit-report/lib/index.js
'use strict'

const reporters = {
  install: require('npm-audit-report/lib/reporters/install'),
  detail: require('npm-audit-report/lib/reporters/detail'),
  json: require('npm-audit-report/lib/reporters/json'),
  quiet: require('npm-audit-report/lib/reporters/quiet'),
}

const exitCode = require('npm-audit-report/lib/exit-code.js')

module.exports = Object.assign((data, options = {}) => {
  const {
    reporter = 'install',
    chalk,
    unicode = true,
    indent = 2,
  } = options

  // CLI defaults this to `null` so the defaulting method above doesn't work
  const auditLevel = options.auditLevel || 'low'

  if (!data) {
    throw Object.assign(
      new TypeError('ENOAUDITDATA'),
      {
        code: 'ENOAUDITDATA',
        message: 'missing audit data',
      }
    )
  }

  if (typeof data.toJSON === 'function') {
    data = data.toJSON()
  }

  return {
    report: reporters[reporter](data, { chalk, unicode, indent }),
    exitCode: exitCode(data, auditLevel),
  }
}, { reporters })
